diff --git a/config.def.h b/config.def.h
index be36294..d934cd6 100644
--- a/config.def.h
+++ b/config.def.h
@@ -6,6 +6,7 @@
 #define MAX_BRIGHTNESS "/sys/class/backlight/amdgpu_bl1/max_brightness"
 #define CAPTURE "Capture"
 #define PLAYBACK "Master"
+#define CPU "/proc/stat"
 #define STEP 5
 #define PFXLEN 4    /* longest pfx + 1 */
 #define MAXLEN 2048 /* more than BLOCKLEN * number of blocks */
@@ -18,4 +19,5 @@ static const Block blocks[] = {
     { CmdBri,  0,  3, "B: " },
     { CmdMic,  0,  4, "M: " },
     { CmdSpk,  0,  5, "S: " },
+    { CmdCpu,  2,  1, "C: " },
 };
diff --git a/dsblocks.c b/dsblocks.c
index cf11ab1..a8f9873 100644
--- a/dsblocks.c
+++ b/dsblocks.c
@@ -15,7 +15,8 @@ enum { CmdDate,
        CmdBat,
        CmdBri,
        CmdMic,
-       CmdSpk };
+       CmdSpk,
+       CmdCpu };
 
 enum { BtnLeft = 1,
        BtnMid,
@@ -32,6 +33,7 @@ typedef struct {
 
 #include "config.h"
 
+#include "cpu.c"
 #include "alsa.c"
 #include "bat.c"
 #include "bri.c"
@@ -42,9 +44,10 @@ static float rbri, maxbri;
 static long rspk, minspk, maxspk;
 static long rmic, minmic, maxmic;
 static int spkon, micon;
-static float briperc, spkperc, micperc;
+static float briperc, spkperc, micperc, cpuperc;
 static char sblocks[LENGTH(blocks)][BLOCKLEN];
 static int counter, debug, update, error;
+static unsigned long long int cput1, cput2, cpui1, cpui2;
 
 void updatebriperc()
 {
@@ -62,6 +65,20 @@ void updatemicperc()
   micperc = 100.0 * (rmic - minmic) / (maxmic - minmic);
 }
 
+void updatecpuperc()
+{
+  if (cput2 && cpui2) {
+    cpuperc = (1.0f - (float)(cpui2 - cpui1) / (float)(cput2 - cput1)) * 100.0;
+    cpui1 = cpui1 ^ cpui2;
+    cpui2 = cpui1 ^ cpui2;
+    cpui1 = cpui1 ^ cpui2;
+
+    cput1 = cput1 ^ cput2;
+    cput2 = cput1 ^ cput2;
+    cput1 = cput1 ^ cput2;
+  }
+}
+
 void cmddate(int i)
 {
   char tmp[DATELEN];
@@ -170,6 +187,19 @@ void cmdspk(int i, int b)
         blocks[i].signal, blocks[i].pfx, blocks[i].signal);
 }
 
+void cmdcpu(int i)
+{
+  if (!cput1 || !cpui1)
+    readcpu(&cput1, &cpui1);
+  else
+    readcpu(&cput2, &cpui2);
+
+  updatecpuperc();
+  snprintf(sblocks[i], sizeof(sblocks[i]),
+      "%c%s%.f%%%c",
+      blocks[i].signal, blocks[i].pfx, cpuperc, blocks[i].signal);
+}
+
 void runcmd(int i, int b)
 {
   switch (blocks[i].cmd) {
@@ -188,6 +218,9 @@ void runcmd(int i, int b)
   case CmdSpk:
     cmdspk(i, b);
     break;
+  case CmdCpu:
+    cmdcpu(i);
+    break;
   }
 }
 
@@ -214,6 +247,13 @@ void setup()
   sigemptyset(&sa.sa_mask);
   update = 1;
   for (int i = 0; i < LENGTH(blocks); i++) {
+    /* min cpu interval is 2 */
+    if (blocks[i].cmd == CmdCpu && blocks[i].interval == 1) {
+      printf("min interval for cpu is 2, aborting\n");
+      error = 1;
+      break;
+    }
+
     if (!blocks[i].signal) {
       printf("use 1 to ignore signal, aborting\n");
       error = 1;
